nifi-nar-bundles/nifi-ccda-bundle/nifi-ccda-processors/src/main/java/org/apache/nifi/processors/ccda/ExtractCCDAAttributes.java

```diff
@@ -0,0 +1,383 @@
+    private JexlEngine jexl = null; // JEXL Engine to execute code for mapping
+    private JexlContext jexlCtx = null; // JEXL Context to hold element being processed
+
+    private List<PropertyDescriptor> properties;
+    private Set<Relationship> relationships;
+
+    /**
+     * SKIP_VALIDATION - Indicates whether to validate the CDA document after loading.
+     * if true and the document is not valid, then ProcessException will be thrown
+     */
+    public static final PropertyDescriptor SKIP_VALIDATION = new PropertyDescriptor.Builder().name("skip-validation")
+            .displayName("Skip Validation").description("Whether or not to validate CDA message values").required(true)
+            .allowableValues("true", "false").defaultValue("true").addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    /**
+     * PRETTY_PRINTING - Choice of whether generated JSON in the FileFlow content is
+     * pretty printed with new line and tabs or just a continuous string of characters
+     */
+    public static final PropertyDescriptor PRETTY_PRINTING = new PropertyDescriptor.Builder().name("pretty-printing")
+            .displayName("Pretty Printing").description("Whether or not to Pretty Print JSON output").required(true)
+            .allowableValues("true", "false").defaultValue("false").addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    /**
+     * REL_SUCCESS - Value to be returned in case the processor succeeds
+     */
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("A FlowFile is routed to this relationship if it is properly parsed as CDA and its content stored in FlowFile")
+            .build();
+
+    /**
+     * REL_FAILURE - Value to be returned in case the processor fails
+     */
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if it cannot be parse CDA and store content to FlowFile.")
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        getLogger().info("Loading packages");
+
+        relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+
+        properties = new ArrayList<>();
+        properties.add(SKIP_VALIDATION);
+        properties.add(PRETTY_PRINTING);
+
+        long start = System.currentTimeMillis();
+
+        // Load required MDHT packages
+        System.setProperty( "org.eclipse.emf.ecore.EPackage.Registry.INSTANCE",

```

Comment:
Setup like this is usually put in a method with the @onScheduled annotation and init(). See [ScanAttribute.java](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ScanAttribute.java#L144) for an example.